python/lsst/pipe/tasks/postprocess.py

```diff
@@ -789,6 +791,167 @@ def _makeArgumentParser(cls):
+    """Task to consolidate per-detector visit metadata
+    """
+    _DefaultName = "consolidateVisitSummary"
+    ConfigClass = ConsolidateVisitSummaryConfig
+
+    @classmethod
+    def _makeArgumentParser(cls):
+        parser = ArgumentParser(name=cls._DefaultName)
+
+        parser.add_id_argument("--id", "calexp",
+                               help="data ID, e.g. --id visit=12345",
+                               ContainerClass=VisitDataIdContainer)
+        return parser
+
+    def writeMetadata(self, dataRef):
+        """No metadata to write.
+        """
+        pass
+
+    def writeConfig(self, butler, clobber=False, doBackup=True):
+        """No config to write.
+        """
+        pass
+
+    def runDataRef(self, dataRefList):
+        self.isGen3 = False
+
+        visit = dataRefList[0].dataId['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefList), visit))
+
+        expCatalog = self.run(visit, dataRefList)
+
+        dataRefList[0].put(expCatalog, 'visitSummary', visit=visit)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        self.isGen3 = True
+
+        dataRefs = butlerQC.get(inputRefs.calexp)
+        visit = dataRefs[0].dataId.byName()['visit']
+
+        self.log.debug("Concatenating metadata from %d per-detector calexps (visit %d)" %
+                       (len(dataRefs), visit))
+
+        expCatalog = self.run(visit, dataRefs)
+
+        butlerQC.put(expCatalog, outputRefs.visitSummary)
+
+    def run(self, visit, dataRefs):
+        """Make a combined exposure catalog from a list of dataRefs.
+
+        Parameters
+        ----------
+        visit : `int`
+            Visit identification number
+        dataRefs : `list`
+            List of calexp dataRefs in visit.  May be list of
+            `lsst.daf.persistence.ButlerDataRef` (Gen2) or
+            `lsst.daf.butler.DeferredDatasetHandle` (Gen3).
+
+        Returns
+        -------
+        visitSummary : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with per-detector summary information.
+        """
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField('visit', type='I', doc='Visit number')
+        schema.addField('ccd', type='I', doc='Detector number')
+        schema.addField('filterName', type='String', size=20, doc='Filter name')
+        schema.addField('psfSigma', type='F',
+                        doc='PSF model full width half-maximum (center of chip)')
+        schema.addField('psfArea', type='F',
+                        doc='PSF model effective area (center of chip)')
+        schema.addField('psfIxx', type='F',
+                        doc='PSF model Ixx (center of chip)')
+        schema.addField('psfIyy', type='F',
+                        doc='PSF model Iyy (center of chip)')
+        schema.addField('psfIxy', type='F',
+                        doc='PSF model Ixy (center of chip)')
```

Comment:
Do these PSF things have units? "counts", maybe, or is it something else?

Reply:
Good question.  I think pixel**2?  I'll confirm.